aip/cloud/2510.md

```diff
@@ -67,6 +67,46 @@ considered PII and user data, but project numbers are not.
 Therefore, when an internal service calls an external Google APIs, it
 **should** use project numbers for making API requests.
 
+### Resource References
+
+Project IDs also appear in [resource names][], which are stored in other
+resources when they are referenced ([example][]).
+
+In these situations, the value of the resource reference should be preserved
+regardless if the user has sent a project ID or number to identify the main
+resource.
+
+For example, for an existing resource book:
+
+```proto
+book {
+  // project-id: "foo"
+  // project-number: 12345
+  name: "project/12345/books/game-of-tones"
+  shelf: "project/123456/shelves/top-row"
+}
+```
+
+A request for the book in the format of `projects/foo/books/game-of-tones`
+should return:
+
+```proto
+book {
+  // Returning the project ID, since that was what was sent.
+  name: "project/foo/books/game-of-tones"
+  // The shelf still uses the project Number, since that is what
+  // the user has sent. If the user has set a name with the project ID,
+  // that should also be preserved.
+  shelf: "project/123456/shelves/top-row"
+}
+```
+
+Similar to how the guidance around project ID / project number is designed
+to ensure that there is a minimal difference between user input and output,
+the reference value is also untranslated to minimize normalized user input.
```

Comment:
"reference value" seems like a term that might be defined. What does it mean?

Reply:
I think "resource reference value" might be clearer. I'll fix.